File: server/src/main/java/com/fluffy/comment/ui/dto/CreateExamCommentWebRequest.java
package com.fluffy.comment.ui.dto;

import com.fluffy.comment.application.dto.CreateExamCommentRequest;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Positive;

public record CreateExamCommentWebRequest(
        @NotBlank String content,
        @Positive Long parentCommentId
) {

    public CreateExamCommentRequest toAppRequest(Long examId, Long memberId) {
        return new CreateExamCommentRequest(content, examId, memberId, parentCommentId);
    }
}

File: server/src/main/kotlin/com/fluffy/comment/ui/request/CreateExamCommentWebRequest.kt
package com.fluffy.comment.ui.request

import com.fluffy.comment.application.request.CreateExamCommentRequest
import jakarta.validation.constraints.NotBlank

data class CreateExamCommentWebRequest(
    @field:NotBlank val content: String,
    val parentCommentId: Long?
) {

    fun toAppRequest(examId: Long, memberId: Long): CreateExamCommentRequest {
        return CreateExamCommentRequest(
            content = content,
            examId = examId,
            memberId = memberId,
            parentCommentId = parentCommentId
        )
    }
}
